pkg/scaffold/controller/controller.go

```diff
@@ -68,6 +72,11 @@ func (a *Controller) GetInput() (input.Input, error) {
 		a.GroupDomain = a.Resource.Group + "." + a.Domain
 	}
 
+	if a.Plural == "" {
+		rs := inflect.NewDefaultRuleset()
+		a.Plural = rs.Pluralize(strings.ToLower(a.Resource.Kind))
```

Comment:
I guess you need a `Kind` to `Resources` converter.
https://github.com/kubernetes/kubernetes/blob/a30f4591609301bd6160abeb22731f70e7dae2e0/staging/src/k8s.io/apimachinery/pkg/api/meta/restmapper.go#L123
This function has been used in several places in k/k.

Reply:
Thanks for the tip. There are couple of places where we mapping a kind to a resource in controller-tools. I can use the recommended way in a separate PR for all the places to keep it consistent.